src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -155,43 +189,131 @@ pub struct RSIndexResult {
             field_mask: 0,
             freq: 0,
             offsets_sz: 0,
             data: RSIndexResultData {
-                num: ManuallyDrop::new(RSNumericRecord(num)),
+                virt: ManuallyDrop::new(RSVirtualResult),
             },
-            result_type: RSResultType::Numeric,
+            result_type: RSResultType::Virtual,
             is_copy: false,
-            metrics: std::ptr::null_mut(),
+            metrics: ptr::null_mut(),
             weight: 0.0,
         }
     }
 
-    /// Create a new virtual index result
-    pub fn virt(doc_id: t_docId) -> Self {
+    /// Create a new numeric index result with the given number
+    pub fn numeric(num: f64) -> Self {
         Self {
-            doc_id,
-            dmd: std::ptr::null(),
-            field_mask: 0,
-            freq: 0,
-            offsets_sz: 0,
+            field_mask: RS_FIELDMASK_ALL,
+            freq: 1,
             data: RSIndexResultData {
-                virt: ManuallyDrop::new(RSVirtualResult),
+                num: ManuallyDrop::new(RSNumericRecord(num)),
             },
-            result_type: RSResultType::Virtual,
-            is_copy: false,
-            metrics: std::ptr::null_mut(),
-            weight: 0.0,
+            result_type: RSResultType::Numeric,
+            weight: 1.0,
+            ..Default::default()
         }
     }
 
+    /// Create a new metric index result
+    pub fn metric() -> Self {
+        Self {
+            field_mask: RS_FIELDMASK_ALL,
+            data: RSIndexResultData {
+                num: ManuallyDrop::new(RSNumericRecord(0.0)),
+            },
+            result_type: RSResultType::Metric,
+            weight: 1.0,
+            ..Default::default()
+        }
+    }
+
+    /// Create a new intersection index result with the given capacity
+    pub fn intersect(cap: usize) -> Self {
+        Self {
+            data: RSIndexResultData {
+                agg: ManuallyDrop::new(RSAggregateResult::with_capacity(cap)),
+            },
+            result_type: RSResultType::Intersection,
+            ..Default::default()
+        }
+    }
+
+    /// Create a new union index result with the given capacity
+    pub fn union(cap: usize) -> Self {
+        Self {
+            data: RSIndexResultData {
+                agg: ManuallyDrop::new(RSAggregateResult::with_capacity(cap)),
+            },
+            result_type: RSResultType::Union,
+            ..Default::default()
+        }
+    }
+
+    /// Create a new hybrid metric index result
+    pub fn hybrid_metric() -> Self {
+        Self {
+            data: RSIndexResultData {
+                agg: ManuallyDrop::new(RSAggregateResult::with_capacity(2)),
+            },
+            result_type: RSResultType::HybridMetric,
+            weight: 1.0,
+            ..Default::default()
+        }
+    }
+
+    /// Create a new term index result with the given term pointer
+    pub fn term(term: *mut RSQueryTerm) -> Self {
```

Comment:
```suggestion
    pub fn term(term: *const RSQueryTerm) -> Self {
```

Reply:
The aim was to have a seamless replacement for this

https://github.com/RediSearch/RediSearch/blob/f232aba9844504d1433716217cf77b841945a216/src/index_result.c#L52-L68

But I see only two of the callers pass in a `term` so defaulting to `null` makes more sense.